storage-api/src/test/java/com/linagora/calendar/storage/ResourceDAOContract.java

```diff
@@ -0,0 +1,366 @@
+            List.of(new ResourceAdministrator(new OpenPaaSId("admin2"), "user")),
+            new OpenPaaSId("creator2"),
+            !DELETED,
+            "desc2",
+            new OpenPaaSId("domain2"),
+            "icon2.png",
+            "Resource2",
+            Instant.now(),
+            Instant.now(),
+            "resource"
+        );
+        ResourceId id1 = dao().insert(req1).block();
+        ResourceId id2 = dao().insert(req2).block();
+        List<Resource> actual = dao().findByDomain(new OpenPaaSId("domain2")).collectList().block();
+
+        assertThat(actual).containsOnly(Resource.from(id2, req2));
+    }
+
+    @Test
+    default void updateShouldUpdateCurrentResource() {
+        ResourceInsertRequest request = new ResourceInsertRequest(
+            List.of(new ResourceAdministrator(new OpenPaaSId("admin1"), "user")),
+            new OpenPaaSId("creator1"),
+            !DELETED,
+            "desc",
+            new OpenPaaSId("domain1"),
+            "icon.png",
+            "ResourceName",
+            Instant.now(),
+            Instant.now(),
+            "resource"
+        );
+        ResourceId resourceId = dao().insert(request).block();
+        ResourceUpdateRequest updateRequest = new ResourceUpdateRequest(
+            Optional.of("UpdatedName"),
+            Optional.of("UpdatedDesc"),
+            Optional.of("updatedIcon.png"),
+            Optional.of(List.of(new ResourceAdministrator(new OpenPaaSId("admin2"), "user")))
+        );
+        Resource updated = dao().update(resourceId, updateRequest).block();
+
+        Resource expected = new Resource(
+            resourceId,
+            List.of(new ResourceAdministrator(new OpenPaaSId("admin2"), "user")),
+            new OpenPaaSId("creator1"),
+            !DELETED,
+            "UpdatedDesc",
+            new OpenPaaSId("domain1"),
+            "updatedIcon.png",
+            "UpdatedName",
+            updated.creation(),
+            updated.updated(),
+            "resource"
+        );
+
+        assertThat(updated).isEqualTo(expected);
+    }
+
+    @Test
+    default void softDeleteShouldMarkResourceAsDeleted() {
+        ResourceInsertRequest request = new ResourceInsertRequest(
+            List.of(new ResourceAdministrator(new OpenPaaSId("admin1"), "user")),
+            new OpenPaaSId("creator1"),
+            !DELETED,
+            "desc",
+            new OpenPaaSId("domain1"),
+            "icon.png",
+            "ResourceName",
+            Instant.now(),
+            Instant.now(),
+            "resource"
+        );
+        ResourceId resourceId = dao().insert(request).block();
+        dao().softDelete(resourceId).block();
+        Resource deleted = dao().findById(resourceId).block();
+
+        assertThat(deleted.deleted()).isTrue();
+    }
+
+    @Test
+    default void searchShouldReturnMatchingResources() {
+        ResourceInsertRequest req1 = new ResourceInsertRequest(
```

Comment:
Can add more test cases for search api?
- which field will be used for matching? name, description, type, id...
- support ignore case?